### PR TITLE
fix(neotest): add neodev integration

### LINKED ISSUE #406: Add neodev integration for neotest community pack
The [neotest README](https://github.com/nvim-neotest/neotest#configuration) says:

>It is highly recommended to use [neodev.nvim](https://github.com/folke/neodev.nvim) to enable type checking for neotest to get type checking, documentation and autocompletion for all API functions.
>
> ```lua
> require("neodev").setup({
>   library = { plugins = { "neotest" }, types = true },
>   ...
> })
> ```

Could we please add this to the astrocommunity's neotest pack [here](https://github.com/AstroNvim/astrocommunity/blob/main/lua/astrocommunity/test/neotest/init.lua)?

### PR DESCRIPTION
**Pull request by Uzaaft**
Should fix #406

### CODE REVIEW

File: lua/astrocommunity/test/neotest/init.lua [hunk 1]
```diff
@@ -5,6 +5,13 @@ return {
     "nvim-neotest/neotest-go",
     "nvim-neotest/neotest-python",
     "rouge8/neotest-rust",
+    {
+      "folke/neodev.nvim",
+      opts = function(_, opts)
+        require("astronvim.utils").list_insert_unique(opts.library.plugins, "neotest")
```
Comment (owittek) [inline lua/astrocommunity/test/neotest/init.lua:11]:
```suggestion
        opts.library.plugins = require("astronvim.utils").list_insert_unique(opts.library.plugins, "neotest")
```

might want to use the "new" syntax for that even though the old one still works

**Review by owittek — CHANGES_REQUESTED** [review]

**Review by owittek — APPROVED** [review]
👍

**Comment by Uzaaft** [comment]
@owittek While your changes seemed correct, I'm not sure they are? neotest seems broken now for me

**Comment by owittek** [comment]
> @owittek While your changes seemed correct, I'm not sure they are? neotest seems broken now for me

Sounds odd as both versions should work the exact same way. Could you confirm that the issue is related to this line?

**Comment by khaneliman** [comment]
After updating my plugins I get this.

```
`Failed to run `config` for neodev.nvim

.../astrocommunity/lua/astrocommunity/test/neotest/init.lua:11: attempt to index field 'library' (a nil value)

# stacktrace:
  - /astrocommunity/lua/astrocommunity/test/neotest/init.lua:11 _in_ **values**
  - ~/.config/nvim/lua/astronvim/utils/lsp.lua:404 _in_ **config**
  - ~/.config/nvim/lua/astronvim/utils/lsp.lua:86 _in_ **setup**
  - ~/.config/nvim/lua/plugins/lsp.lua:45
  - /mason-lspconfig.nvim/lua/mason-lspconfig/init.lua:84 _in_ **fn**
  - /mason.nvim/lua/mason-core/optional.lua:101 _in_ **if_present**
  - /mason-lspconfig.nvim/lua/mason-lspconfig/init.lua:82 _in_ **fn**
  - /mason.nvim/lua/mason-core/functional/list.lua:98 _in_ **each**
  - /mason-lspconfig.nvim/lua/mason-lspconfig/init.lua:92 _in_ **setup_handlers**
  - /mason-lspconfig.nvim/lua/mason-lspconfig/init.lua:50 _in_ **setup**
  - ~/.config/nvim/lua/plugins/configs/mason-lspconfig.lua:2 _in_ **config**
  - ~/.config/nvim/lua/astronvim/utils/init.lua:144
 ```

**Comment by Uzaaft** [comment]
@khaneliman I'm on it. :)

**Pull request merged**